lib/pmap.dart

```diff
@@ -1,109 +1,178 @@
 import 'dart:async';
 import 'dart:isolate';
 
-class _Enumerated<T> {
-  final int index;
-  final T value;
-  _Enumerated({this.index, this.value});
+/// The *main* process for the spawned isolates.
+void _process(_Processor processor) async {
+  final receivePort = ReceivePort();
+  processor.sendPort.send(receivePort.sendPort);
+  await for (dynamic input in receivePort) {
+    processor.process(input);
+  }
 }
 
+/// This [_Processor] is given to the spawned isolates as the work basis.
 class _Processor<T, U> {
-  SendPort sendPort;
-  U Function(T input) mapper;
-  int sendCount = 0;
+  final SendPort sendPort;
+  final U Function(T input) mapper;
 
   void process(dynamic input) async {
-    _Enumerated<T> enumeratedInput = input;
-    sendPort.send(_Enumerated(index:enumeratedInput.index, value:mapper(enumeratedInput.value)));
+    MapEntry<int, T> enumeratedInput = input;
+    sendPort.send(MapEntry(enumeratedInput.key, mapper(enumeratedInput.value)));
   }
-}
 
-void _process(SendPort sendPort) async {
-  var receivePort = new ReceivePort();
-  sendPort.send(receivePort.sendPort);
-  _Processor processor;
-  await for (dynamic input in receivePort) {
-    if (processor == null) {
-      processor = input;
-      processor.sendPort = sendPort;
-    } else {
-      processor.process(input);
-    }
-  }
+  _Processor(this.mapper, this.sendPort);
 }
 
+/// The data to manage a spawned [Isolate]
```

Comment:
How about:
/// An isolate that will execute the `mapper` function on all items it receives via the `receivePort`.